srml/support/src/storage/hashed/generator.rs

```diff
@@ -202,16 +203,54 @@ pub trait StorageValue<T: codec::Codec> {
 	/// Append the given items to the value in the storage.
 	///
 	/// `T` is required to implement `codec::EncodeAppend`.
+	///
+	/// This function only exists for storage values without default value..
 	fn append<S: HashedStorage<Twox128>, I: codec::Encode>(
-		items: &[I], storage: &mut S
-	) -> Result<(), &'static str> where T: codec::EncodeAppend<Item=I> {
+		items: &[I],
+		storage: &mut S,
+	) -> Result<(), &'static str>
+		where T: codec::EncodeAppend<Item=I>, Self: crate::traits::NoDefault,
```

Comment:
To avoid the confusion of
```
pub foo: Vec<u32> = vec![1]; 
assert__eq!(foo::append(&[2]), vec![1,2]); 
assert__eq!(foo::append(&[2]), vec![2])
```